lesson22/src/js/main.js

```diff
@@ -0,0 +1,234 @@
+    {state:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {state:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortCategories = ["id", "age"];
+
+const createSortButtons = (columnKey) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle js-sortButtons-Box"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtonAttributes){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-sortButtons-Item`,
+                'data-state':button.state
+            }
+        });
+
+        button.state !== "both" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+
+    sortButtonsBox.appendChild(fragment);
+    return sortButtonsBox;
+}
+
+const toggleHiddenClassForSortButton = (currentButton,nextButton) => {
+    currentButton.classList.add("hidden");
+    nextButton.classList.remove("hidden");
+}
+
+const updateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (category, usersData, state) => {
+    const cloneUsersData = [...usersData];
+
+    switch (state) {
+        case "asc":
+            cloneUsersData.sort((a, b) => a[category] - b[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        case "desc":
+            cloneUsersData.sort((a, b) => b[category] - a[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        default:
+            updateTableBody(usersData);
+    }
+}
+
+const changeState = (currentState) => {
+    switch (currentState) {
+      case "asc":
+        return "desc";
+      case "desc":
+        return "both";
+      default:
+        return "asc";
+    }
+}
+
+const addIsNotClickedForSortButtonsBox = (currentTarget) => {
+    const sortButtons = document.querySelectorAll(".js-sortButtons-Box");
+    for(const button of sortButtons){
+        button.classList.add("is-not-clicked");
+    }
+
+    currentTarget.classList.remove("is-not-clicked");
+}
+
+const resetStatusForSortButton = () => {
+    const sortButtons = document.querySelectorAll(".is-not-clicked > .js-sortButtons-Item");
+    sortButtons.forEach((button) => {
+        const state = button.getAttribute("data-state");
+        (state !== "both") ? button.classList.add("hidden") : button.classList.remove("hidden");
+    });
+}
```

Comment:
I've added a function to restore the status to default except for the button you clicked. 
However, I feel it is redundant.

Reply:
Sorry I can't help you. When I did this assignment, I implemented the same logic.😫